haproxy_discovery.sh

```diff
@@ -11,7 +11,7 @@
 #  global
 #  stats socket /run/haproxy/info.sock  mode 666 level user
 
-HAPROXY_SOCK=$(echo $1 | grep -Po "^/[^\s]+" || echo "/var/run/haproxy/info.sock")
+HAPROXY_SOCK=$(echo $1 | tr -d '\040\011\012\015' || echo "/var/run/haproxy/info.sock")
```

Comment:
Yes, this is working perfectly. Sorry about not regression-testing the without parameter functionality.